### PR TITLE
PE-567: Ensure no fraction of Winston

### PR DESCRIPTION
**Pull request by matibat**

### CODE REVIEW

**Comment by matibat** [comment]
It will be a draft until I push the commit for printing only 12 digits in the `get-balance` command

**Review by arielmelendez — CHANGES_REQUESTED** [review]
Minor comments and Unit Test please.

File: src/commands/send_ar.ts [hunk 1]
```diff
@@ -14,13 +14,19 @@ new CLICommand({
...
+			throw new Error(`The AR amount must have a maximum of 12 digits of precision`);
+		}
+		const arAmount: number = +options.arAmount;
+		const destAddress: string = options.destAddress;
 		const wallet = await parameters.getRequiredWallet();
 		const walletAddress = await wallet.getAddress();
 		console.log(walletAddress);
-		console.log(`arAmount: ${options.arAmount}`);
-		console.log(`destAddress: ${options.destAddress}`);
+		console.log(`arAmount: ${arAmount.toFixed(12)}`);
```
[the remark this answers is not in the document]
Comment (arielmelendez) [reply src/commands/send_ar.ts:26]: Newline buffering might be nice too

**Comment by matibat** [comment]
After adding the tests, I realized the function was failing for valid integer values. The approach now is to match the string with a RegExp looking for the numbers after the comma, but before the trailing zeros at the end:
```ts
/**
 * Capturing group 1: the points after the comma with trimmed zeros
 */
export const FLOATING_POINT_REGEXP = /^\d*(?:\.(\d*[1-9])0*)?$/;
```
Explanation of the regexp:
1. `^` the beginning of the string
2. `\d*` any number of digits
3. `(?: )?` zero or one non-capturing group
4. `\.` the literal point **<---- It does depend in the locale, right?**
5. `( )` the capturing group
6. `\d*` any number of digits
7. `[1-9]` a digit that is explicitly non-zero
8. `0*` none or more zeros
9. `$` end of the string

So...
 - (5) will catch all relevant digits to run the assertion on
 - (1) and (2) makes the match fail for negative numbers
 - (2) lets it to match a floating point number without units (e.g. `.0012`, `.0`)
 - (3) being this group conditional allow it to match for integers

Then, anything matching this pattern is a valid string representation of a positive number. If it doesn't match then it is not a number or it is negative.

**Comment by matibat** [comment]
**EDIT**

Added a case for this specific example: `.00000000` on which will match a single `0`

**Comment by arielmelendez** [comment]
> 4\. It does depend in the locale, right?

**Pull request merged**